tests/parametric/test_span_links.py

```diff
@@ -0,0 +1,189 @@
+        assert span_has_no_parent(span)
+        assert span["meta"].get(ORIGIN) is None
+        assert span["meta"].get("_dd.p.dm") != "-4"
+        assert span["metrics"].get(SAMPLING_PRIORITY_KEY) != 2
+
+        assert span.get("span_links") is not None
+
+        span_links = span["span_links"]
+        assert len(span_links) == 1
+        link = span_links[0]
+        assert link.get("span_id") == 9876543210
+        assert link.get("trace_id") == 1234567890
+        assert link.get("trace_id_high") == 16
+
+        assert link.get("tracestate") is not None
+        tracestateArr = link["tracestate"].split(",")
+        assert len(tracestateArr) == 1 and tracestateArr[0].startswith("dd=")
+        tracestateDD = tracestateArr[0][3:].split(";")
+        assert len(tracestateDD) == 3
+        assert "o:synthetics" in tracestateDD
+        assert "s:2" in tracestateDD
+        assert "t.dm:-4" in tracestateDD
+
+        assert (link.get("flags") or 0) == 0
+        assert len(link.get("attributes")) == 2
+        assert link["attributes"].get("foo") == "bar"
+        assert link["attributes"].get("_dd.p.dm") == "-4"
+
+    def test_span_link_from_distributed_w3c_headers(self, test_agent, test_library):
+        """Properly inject w3c distributed tracing information into span links.
+        This mostly tests that the injected tracestate and flags are accurate.
+        """
+        with test_library:
+            with test_library.start_span(
+                "root",
+                links=[Link(parent_id=0)],
+                http_headers=[
+                    ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                    ["tracestate", "foo=1,dd=t.dm:-4;s:2"],
+                ],
+            ):
+                pass
+
+        span = get_span(test_agent)
+        assert span.get("trace_id") != 1234567890
+        assert span_has_no_parent(span)
+        assert span["meta"].get("_dd.p.dm") != "-4"
+        assert span["metrics"].get(SAMPLING_PRIORITY_KEY) != 2
+
+        assert span.get("span_links") is not None
+
+        span_links = span["span_links"]
+        assert len(span_links) == 1
+        link = span_links[0]
+        assert link.get("span_id") == 1311768467284833366
+        assert link.get("trace_id") == 8687463697196027922
+        assert link.get("trace_id_high") == 1311768467284833366
+
+        assert link.get("tracestate") is not None
+        tracestateArr = link["tracestate"].split(",")
+        assert len(tracestateArr) == 2 and tracestateArr[0].startswith("dd=")
+        assert tracestateArr[1] == "foo=1"
+        tracestateDD = tracestateArr[0][3:].split(";")
+        assert len(tracestateDD) == 2
+        assert "s:2" in tracestateDD
+        assert "t.dm:-4" in tracestateDD
+
+        assert link.get("flags") == 1 | -2147483648
+        assert len(link.get("attributes")) == 1
+        assert link["attributes"].get("_dd.p.dm") == "-4"
```

Comment:
I don't see why there should a `"_dd.p.dm"` attribute. Can you clarify why that's there?

Reply:
Let me remove it, it's anyway in the tracestate for whomever needs that. And the RFC doesn't say that we should extract datadog tags into the attributes.